*: AWS DMS Source Database prototype

otan opened the pull request:
Refs: https://github.com/cockroachdb/cockroach/pull/84505

otan committed:
disable WITH HOLD unimplemented check

otan committed:
mock pg_timezone_names, pg_settings for log_timezone

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/93404)
<!-- Reviewable:end -->

otan committed:
begin prototype of replication slots

otan committed:
add ability to drop a changefeed

otan committed:
somewhat parse replication options

ajstorm committed:
first hacky attempt at CDS record parsing

- Still hard codes table name
- Worse, hard codes types, which is just broken

Epic: none

Release note: None

otan committed:
implement replication protocol

otan committed:
impl db_replication_slots

otan committed:
make more stuff work

otan committed:
fake aws dms setup

otan committed:
identify system, slots, fake more stuff

ajstorm committed:
Cleanup a couple of loose ends

- Handle multiple tables in the database
- Remove hard-coding of table and type names

Epic: none

Release note: None

ajstorm committed:
end otan sadness

Add back in (a correct) schema name

Epic: none

Release note: None

ajstorm committed:
remove errant comma

Epic: none

Release note: None

otan committed:
make work for DMS by fixing standard name, removing DB hacks, drop slot
